src/MerchantCenter/MerchantStatuses.php

```diff
@@ -844,33 +853,23 @@ protected function update_products_meta_with_mc_status() {
 				}
 			}
 		}
-		ksort( $new_product_statuses );
-
-		/** @var ProductMetaQueryHelper $product_meta_query_helper */
-		$product_meta_query_helper = $this->container->get( ProductMetaQueryHelper::class );
-
-		// Get all MC statuses.
-		$current_product_statuses = $product_meta_query_helper->get_all_values( ProductMetaHandler::KEY_MC_STATUS );
-
-		// Format: product_id=>status.
-		$to_insert = [];
-		// Format: status=>[product_ids].
-		$to_update = [];
 
 		foreach ( $new_product_statuses as $product_id => $new_status ) {
-			if ( ! isset( $current_product_statuses[ $product_id ] ) ) {
-				// MC status not in WC, insert.
-				$to_insert[ $product_id ] = $new_status;
-			} elseif ( $current_product_statuses[ $product_id ] !== $new_status ) {
-				// MC status not same as WC, update.
-				$to_update[ $new_status ][] = intval( $product_id );
+			$product = $products[ $product_id ] ?? null;
+
+			// At this point, the product should exist in WooCommerce but in the case that product is not found, log it.
+			if ( ! $product ) {
+				do_action(
+					'woocommerce_gla_debug_message',
+					sprintf( 'Merchant Center product %s not found in this WooCommerce store.', $product_id ),
```

Comment:
This error message seems to implicate that we are outputting the MC ID, but at this time we have already mapped it to a WC ID (potentially with a filter and custom mapping). Should we reword it to something like:
```suggestion
					sprintf( 'Merchant Center product with WooCommerce ID %d is not found in this store.', $product_id ),
```